float32 lat
float32 lng
float32 angle